functions.php

```diff
@@ -333,9 +344,87 @@ public function register_p4_page_type_taxonomy() {
 	}
 
 	/**
-	 * Registers custom post types.
+	 * Save custom taxonomy for planet4 post types
+     *
+     * @param int post_id
 	 */
-	public function register_post_types() {}
+	public function p4_save_post_type( $post_id ) {
+		/**
+         * Some of these checks might be redundant, but they're all nicely
+         * separated and easy to delete so I'll leave them for now.
+         */
+
+        // Ignore autosave.
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+			return;
+		}
+		// Check nonce.
+		if ( ! isset( $_POST['p4-page-type-nonce'] ) || ! wp_verify_nonce( $_POST['p4-page-type-nonce'], basename( __FILE__ ) ) ) {
+			return;
+		}
+		// Check user's capabilities.
+		if ( ! current_user_can( 'edit_post', $post_id ) ) {
+			return;
+		}
+		// Make sure there's input.
+		if ( ! isset( $_POST['p4-page-type'] ) ) { // Input var okay.
+			return;
+		}
+		// If "none" was selected.
+		if ( $_POST['p4-page-type'] === '-1' ) {
+			$this->p4_remove_page_type( $post_id );
+		}
+		// Make sure the term exists and it's not an error.
+        $selected = get_term_by( 'slug', sanitize_text_field( $_POST['p4-page-type'] ), 'p4-page-type' ); // Input var okay.
+
+		if ( false === $selected || is_wp_error( $selected ) ) {
+			return;
+		}
+		// Save post type.
+		wp_set_post_terms( $post_id, $selected->slug, 'p4-page-type', $append = false );
+    }
+	/**
+	 * Removes primary category from a post.
+	 *
+	 * @param $post_id
+	 */
+	public function p4_remove_page_type( $post_id ) {
+		if ( $term = get_the_terms( $post_id, 'p4-page-type' ) ) {
+			if ( ! is_wp_error( $term ) ) {
+				wp_remove_object_terms( $post_id, $term[0]->term_id, 'p4-page-type' );
+			}
+		}
+	}
+	/**
+	 * Add a dropdown to choose planet4 post type.
+     *
+     * @param WP_Post $object
+	 */
+	public function p4_metabox_markup( $object ) {
```

Comment:
Instead of the conditional below, you could do the following which would be cleaner:

`p4_metabox_markup( WP_Post  $object ) {`

This will enforce the correct type coming into this function for you.